components/gcp/container/component_sdk/python/kfp_component/launcher/__main__.py

```diff
@@ -26,9 +27,19 @@ def main():
         description='Launch a python module or file.')
     parser.add_argument('file_or_module', type=str,
         help='Either a python file path or a module name.')
+    parser.add_argument(
+        'ui_metadata_path',
+        type=str,
+        default='/mlpipeline-ui-metadata.json',
+        help='Path for the file where the mlpipeline-ui-metadata.json data '
+             'should be written.')
     parser.add_argument('args', nargs=argparse.REMAINDER)
     args = parser.parse_args()
-    launch(args.file_or_module, args.args)
+
+    if args.ui_metadata_path:
+        os.environ['KFP_UI_METADATA_PATH'] = args.ui_metadata_path
+
+    launch(args.file_or_module, [args.args, args.ui_metadata_path])
```

Comment:
Are you sure the `args.ui_metadata_path` needs to be passed to functions? They usually do no accept it.

Reply:
I've verified it works by building a new container image. But your point is good. The actual root cause might be simply the argument name should be `ui_metadata_path` but not `--ui_metadata_path`.

Let me take a shot to see if that works.